frontend/actions/src/main/java/com/virtuslab/gitmachete/frontend/actions/traverse/TraverseSyncToRemote.java

```diff
@@ -200,8 +203,15 @@ private boolean handleDivergedFromRemote(IManagedBranchSnapshot gitMacheteBranch
         new ResetCurrentToRemoteBackgroundable(
             getString("action.GitMachete.BaseResetToRemoteAction.task-title"),
             /* canBeCancelled */ true, gitMacheteBranch.getName(), remoteTrackingBranch.getName(),
-            gitRepository).queue();
-        return true;
+            gitRepository) {
+          @Override
+          public void onSuccess() {
+            graphTable.queueRepositoryUpdateAndModelRefresh(traverseNextEntry);
+          }
+        }.queue();
+        // The ongoing traverse is now a responsibility of the freshly-queued backgroundable;
+        // NOT a responsibility of the outer method.
+        return false;
```

Comment:
Here's the actual fix...

Reply:
oh, I see. so previously reset could happen after the traverse of a next branch, right?